### ISSUE TITLE
Incomplete function definition "compatible" with int

### ISSUE THREAD

**Issue by ikonst** [python/mypy#15261]
Without `--strict`, this isn't an error:
```python
class A:
    xyz: int


class B(A):
    def xyz(self):
        pass
```

Clearly `xyz` is incomplete, and making it complete results in
```
error: Signature of "xyz" incompatible with supertype "A" 
```
but even without completing it, its type is revealed as `def (self: scratch.B) -> Any` which already rules out compatibility.

**Comment by erictraut** [comment]
@ikonst, I don't understand what you mean. Can you elaborate? What do you mean by "incomplete function definition"? With the code sample above, what error (if any) would you expect mypy to emit? Its current behavior seems correct to me.

**Comment by JelleZijlstra** [comment]
<img width="471" alt="Screenshot 2023-08-14 at 2 23 34 PM" src="https://github.com/python/mypy/assets/906600/661ec32c-6140-4549-b779-165552f20b3a">

The method is "incomplete" because it lacks an annotation. I think @ikonst is saying mypy should emit an error similar to the pyright one in this screenshot; I agree.

**Comment by sterliakov** [comment]
As of 1.15.0, mypy reports this violation when `--strict` (or at least `--check-untyped-defs`) is enabled. In non-strict mode untyped methods aren't checked for LSP compatibility by design.
